crates/sqlbuiltins/src/functions/scalars/mod.rs

```diff
@@ -51,16 +54,127 @@ impl BuiltinScalarUDF for Version {
     }
 }
 
-fn get_nth_scalar_value(input: &[ColumnarValue], n: usize) -> Option<ScalarValue> {
+fn get_nth_scalar_value(
+    input: &[ColumnarValue],
+    n: usize,
+    op: &dyn Fn(ScalarValue) -> Result<ScalarValue, BuiltinError>,
+) -> Result<ColumnarValue, BuiltinError> {
     match input.get(n) {
         Some(input) => match input {
-            ColumnarValue::Scalar(scalar) => Some(scalar.clone()),
-            ColumnarValue::Array(arr) => ScalarValue::try_from_array(arr, 0).ok(),
```

Comment:
this is the bug in the previous implementation: if you got a column reference previously we would only look at the first element in the column and ignore other values. 

The change that follows allows us to process all of the values in the column, using the (newly) provided closure.

Reply:
can’t this simply be:

```
ColumnarValue::Array(arr) => Ok(ColumnarValue::Array(ScalarValue::iter_to_array(
    (0..arr.len()).map(|idx| -> Result<ScalarValue, ExtensionError> {
        Ok(op(ScalarValue::try_from_array(arr, idx)?)?)
    })
)?)),
```